src/pybamm/solvers/processed_variable.py

```diff
@@ -133,19 +133,24 @@ def _setup_cpp_inputs(self, t, full_range):
         ys = self.all_ys
         yps = self.all_yps
         inputs = self.all_inputs_casadi
+
+        # Remove all empty ts
+        idxs_nonempty = np.where([ti.size > 0 for ti in ts])[0]
+
         # Find the indices of the time points to observe
         if full_range:
-            idxs = range(len(ts))
+            idxs = idxs_nonempty
```

Comment:
do we really need this branch of the if statement? why not call it `idx` to begin with?

Reply:
I think I was just looking to avoid redefining `idxs = idxs[idxs_subset]`, but the previous way is probably just more confusing